### PR TITLE
Added EventSubscriberInterface marker interface to support autoconfiguration

### LINKED ISSUE #674: Autoconfiguration
Can we have the `EventSubscriber` autoconfigured?
And it would be also nice if entity listeners can be autoconfigured, maybe by creating a marker interface?

### PR DESCRIPTION
**Pull request by lyrixx**
fixes #674

### CODE REVIEW

**Comment by ostrolucky** [comment]
This is solution for https://github.com/doctrine/DoctrineBundle/issues/821 right? But interface proposed in this PR looks nothing like proposals in issue.

**Comment by lyrixx** [comment]
Nope, it's for https://github.com/doctrine/DoctrineBundle/issues/674 as stated in the PR description.

File: EventSubscriber/EventSubscriberInterface.php [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\EventSubscriber;
+
+interface EventSubscriberInterface
```
Comment (greg0ire) [inline EventSubscriber/EventSubscriberInterface.php]: Shouldn't it extend another interface? That would make it easier on the user.

Comment (lyrixx) [reply EventSubscriber/EventSubscriberInterface.php]: Good idea. Fixed.

**Comment by alcaeus** [comment]
Thanks @lyrixx! 🎉

**Comment by rvanlaak** [comment]
Brilliant in its simplicity! 👏 

Would it be possible to on userland let DI alias the old interface for the new one, so without making code changes to PHP this autowiring could get used (read: tested) on our application?

**Comment by lyrixx** [comment]
@rvanlaak thanks.

If you are using **only** the ORM (and not the ORM) (see explaination [here](https://github.com/doctrine/DoctrineBundle/issues/674#issuecomment-349347553) you can do [that](https://github.com/doctrine/DoctrineBundle/issues/674#issuecomment-503519110):

```
services:
    _instanceof:
        Doctrine\Common\EventSubscriber:
            tags: [ doctrine.event_subscriber ]
```

**Comment by rvanlaak** [comment]
Had that one in mind, which probably is sufficient. Was more thinking of something conceptual like:

```yaml
Doctrine\Common\EventSubscriber:
    alias: Doctrine\Bundle\DoctrineBundle\EventSubscriber\EventSubscriberInterface 
```

This in order to gain advantage of all functionality the doctrine bundle in the possible future would add via DI (e.g. tags, compiler passes) based on `EventSubscriberInterface`.

**Pull request merged**